ENIGMAsystem/SHELL/Universal_System/Extensions/VideoPlayer/videoplayer.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+
+ MIT License
+ 
+ Copyright © 2020 Samuel Venable
+ 
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+ 
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+ 
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+ 
+*/
+
+#include <string>
+#include <thread>
+#include <map>
+
+#include "videoplayer.h"
+
+#include <mpv/client.h>
+
+using std::string;
+
+static std::map<string, string> vidmap;
+static std::map<string, string> widmap;
+
+static std::map<string, string> plymap;
+static std::map<string, string> gfxmap;
+
+static std::map<string, bool>   fllmap;
+static std::map<string, int>    volmap;
```

Comment:
Instead of all these maps, it's safer to use a single map of ID to a struct:
```cpp
struct VideoData {
  mpv_handle *mpv;
  std::string window_id;
  std::string gpu_api;
  int volume;
  bool is_playing;
  bool is_fullscreen;
};
static std::map<std::string, std::string> videos;
```